packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -790,3 +870,48 @@ function getResponseId(responses: QuestionnaireResponseItem[], index: number): s
   }
   return responses[index].id as string;
 }
+
+function formatSelectData(item: QuestionnaireItem): any {
+  if (item.answerOption?.length === 0) {
+    return undefined;
+  }
+  const option = (item.answerOption as QuestionnaireItemAnswerOption[])[0];
+  const optionValue = getTypedPropertyValue(
+    { type: 'QuestionnaireItemAnswerOption', value: option },
+    'value'
+  ) as TypedValue;
+  const propertyName = 'value' + capitalize(optionValue.type);
+
+  const data = (item.answerOption ?? []).map((a) => ({
+    value:
+      propertyName === 'valueCoding' ? a.valueCoding?.code : a[propertyName as keyof QuestionnaireItemAnswerOption],
+    label:
+      propertyName === 'valueCoding' ? a.valueCoding?.display : a[propertyName as keyof QuestionnaireItemAnswerOption],
+  }));
+  return { propertyName, data };
+}
+
+function defaultValueById(
+  answers: Record<string, QuestionnaireResponseItemAnswer[]>,
+  item: QuestionnaireItem,
+  multiple?: boolean
+): any {
+  const results = [];
+  for (const answer in answers) {
+    if (answer === item.id) {
+      for (const answerValue of answers[answer]) {
+        const itemValue = getTypedPropertyValue(
+          { type: 'QuestionnaireItemAnswerOption', value: answerValue },
+          'value'
+        ) as TypedValue;
+
+        if (itemValue?.type === 'Coding') {
```

Comment:
Why is this special case necessary?

Reply:
The defaultValue prop in the NativeSelect doesn't accept the object and wants a string. If we pass in the object it won't show in the next components as we step along the questionnaire. 

Since it might not just be valueCoding and passing in valueString or valueInteger I added this edge case.

Reply:
Hmm, ok, I think this is a bug in `<QuestionnaireFormItem>`.  We should not add a special case here, because we're going to tie ourselves in knots inside `<QuestionnaireFormItem>`.